packages/astro/src/runtime/server/render/astro/render.ts

```diff
@@ -31,6 +31,10 @@ export async function renderToString(
 	let str = '';
 	let renderedFirstPageChunk = false;
 
+	if (isPage) {
+		await bufferHeadContent(result);
+	}
```

Comment:
This was the missing piece to make the non-streaming render work. It exists in `renderToReadableStream` but wasn't ported here too.

Reply:
So simple!